Monika After Story/game/script-greetings.rpy

```diff
@@ -4200,6 +4200,100 @@ label greeting_back_from_hangout:
+
+    addEvent(
+        Event(
+            persistent.greeting_database,
+            eventlabel="greeting_spacing_out",
+            conditional="store.mas_getAbsenceLength() >= datetime.timedelta(hours=3)",
+            unlocked=True,
+            rules=ev_rules,
+            aff_range=(mas_aff.LOVE, None)
+        ),
+        code="GRE"
+    )
+
+    del ev_rules
+
+label greeting_spacing_out:
+    # couple of things:
+    # shield ui
+    $ mas_RaiseShield_core()
+
+    # 3 - keymaps not set (default)
+    # 4 - hotkey buttons are hidden (skip visual)
+    # 5 - music is off (skip visual)
+
+    $ use_right_smug = bool(random.randint(0, 1))
+
+    # Need to do this to force the exp immediately
+    call spaceroom(dissolve_all=True, scene_change=True, force_exp="monika 1gsbsu" if use_right_smug else "monika 1msbsu")
+
+    python hide:
+        # Define some other things we're going to work with
+        spacing_out_pause = PauseDisplayableWithEvents()
+        events = list()
+        next_event_time = 0
+        right_smug = (renpy.partial(renpy.show, "monika 1gsbsu"), renpy.restart_interaction)
+        left_smug = (renpy.partial(renpy.show, "monika 1msbsu"), renpy.restart_interaction)
+
+        # Make the events which will change exps
+        for i in range(random.randint(4, 6)):
+            events.append(
+                PauseDisplayableEvent(
+                    datetime.timedelta(seconds=next_event_time),
+                    right_smug if use_right_smug else left_smug
+                )
+            )
+            next_event_time += random.uniform(0.9, 1.8)
+            use_right_smug = not use_right_smug
+        # The last exp in the sequence
+        events.append(
+            PauseDisplayableEvent(
+                datetime.timedelta(seconds=next_event_time),
+                (renpy.partial(renpy.show, "monika 1tsbsu"), renpy.restart_interaction)
+            )
+        )
+        next_event_time += 0.7
+        # This is to automatically cancel the pause after all the events
+        events.append(
+            PauseDisplayableEvent(
+                datetime.timedelta(seconds=next_event_time),
+                spacing_out_pause.stop
+            )
+        )
+        spacing_out_pause.events[:] = events
+
+        spacing_out_pause.start()
+
+    # Small pause so people don't skip this line
+    $ renpy.pause(0.1)
+    $ enable_esc()
+    m 2wubfsdlo "[player]!"
+    m 1rubfsdlb "You surprised me! {w=0.4}{nw}"
+    extend 1eubsu "I was{w=0.2} spacing out a bit..."
+    m 1hubsb "Ahaha~"
+    m 1eua "I'm very happy to see you again. {w=0.2}{nw}"
+    extend 3eua "What should we do today, [player]?"
+
+    # cleanup
+    python:
+        # enable music menu and music hotkeys
+        mas_MUINDropShield()
+        # 3 - set the keymaps
+        set_keymaps()
+        # 4 - hotkey buttons should be shown
+        HKBShowButtons()
+        # 5 - restart music
+        mas_startup_song()
+
```

Comment:
other visual skip greetings had the problem of quick menu/escape not being enabled, is this one affected, too?

Reply:
It was, but now we use `enable_esc` right before the dlg starts.